### PR TITLE
Federated sharing

### PR DESCRIPTION
**Pull request by nasli**

### CODE REVIEW

File: Owncloud iOs Client/DataBase/Queries/ManageSharesDB.m [hunk 1]
```diff
@@ -630,24 +630,25 @@ + (OCSharedDto *) getTheOCShareByFileDto:(FileDto*)file andShareType:(NSInteger)
...
-+ (void) updateTheRemoteShared: (NSInteger)idRemoteShared withPermissions: (NSInteger)permissions{
++ (void) updateTheRemoteShared: (NSInteger)idRemoteShared forUser: (NSInteger)userId withPermissions: (NSInteger)permissions{
     DLog(@"updateTheFileIDwithPermissions");
     FMDatabaseQueue *queue = Managers.sharedDatabase;
     
     [queue inTransaction:^(FMDatabase *db, BOOL *rollback) {
         BOOL correctQuery=NO;
         
-        correctQuery = [db executeUpdate:@"UPDATE shared SET permissions = ? WHERE id_remote_shared = ?", [NSNumber numberWithInt:permissions], [NSNumber numberWithInteger:idRemoteShared]];
+        correctQuery = [db executeUpdate:@"UPDATE shared SET permissions = ? WHERE id_remote_shared = ? AND user_id = ?", [NSNumber numberWithInt:permissions], [NSNumber numberWithInteger:idRemoteShared], [NSNumber numberWithInteger:userId]];
```
Comment (nasli) [inline Owncloud iOs Client/DataBase/Queries/ManageSharesDB.m]: Better uses numberWithInteger instead of numberWithInt

**Comment by javiergonzper** [comment]
Thank you @owncloud-bot :satisfied:

File: Owncloud iOs Client/Tabs/FileTab/Share/ShareEditUserViewController.m [hunk 2]
```diff
@@ -157,10 +158,10 @@ - (void) didSelectCloseView {
...
 - (void) reloadView {
     
-    if (self.canEditEnabled && self.sharedItem.isDirectory){
-        self.optionsShownWithCanEdit = optionsShownIfFileIsDirectory;
+    if (self.canEditEnabled && self.sharedItem.isDirectory && self.updatedOCShare.shareType != shareTypeRemote){
+        self.optionsShownWithCanEdit = fullOptionsForCanEditOption;
     }else{
-        self.optionsShownWithCanEdit = optionsShownIfFileIsNotDirectory;
+        self.optionsShownWithCanEdit = minOptionsForCanEditOption;
     }
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/FileTab/Share/ShareEditUserViewController.m]: Is optionsShownWithCanEdit used?

File: Owncloud iOs Client/Tabs/FileTab/Share/ShareSearchUserViewController.m [hunk 3]
```diff
@@ -403,7 +427,17 @@ - (void)customizeMessageView:(TSMessageView *)messageView
     messageView.duration = messageDuration;
 }
 
-
+#pragma mark - Fake Federating user
```
Comment (mRoRo) [inline Owncloud iOs Client/Tabs/FileTab/Share/ShareSearchUserViewController.m]: I think it's better not use fake in the comment

**Pull request merged**